normalizer_lib.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+"""Library of language-agnostic FST rewrite rules to normalize text."""
+
+import unicodedata
+from dataclasses import dataclass
+from typing import List
+from pynini import *
+import conf
+from config import *
+
+
+LANGUAGE = conf.LANGUAGE
+
+GRAPHEMES = LANGUAGE.GRAPHEMES
+
+INITIAL_PUNCTUATION = LANGUAGE.INITIAL_PUNCTUATION
+FINAL_PUNCTUATION = LANGUAGE.FINAL_PUNCTUATION
+OTHER_PUNCTUATION = union(r"\[", r"\]")
+PUNCTUATION = union(INITIAL_PUNCTUATION, FINAL_PUNCTUATION, OTHER_PUNCTUATION)
+NUMBERS = LANGUAGE.NUMBERS
+SPACE = acceptor(" ")
+
+SIGMA_STAR = union(*("[{}]".format(i) for i in range(1, 256))
+                   ).optimize().closure()
+
+
+@dataclass
+class Verbalizable:
+    """Contains verbalizable tokens."""
+
+    @staticmethod
+    def verbalizable():
+        """Returns union of verbalizable tokens.
+
+        args: none
+
+        returns: Union of acceptors for verbalizable tokens.
+        """
+        token = (INITIAL_PUNCTUATION.ques +
+                 (GRAPHEMES.plus | NUMBERS.plus).plus +
+                 FINAL_PUNCTUATION.ques)
+
+        email_address = (GRAPHEMES.plus +
+                         "@" +
+                         GRAPHEMES.plus +
+                         closure("." + GRAPHEMES.plus, 1, 2))
+
+        web_address = ((("http" + acceptor("s").ques + "://").ques +
+                        "www.").ques +
+                       union(GRAPHEMES, NUMBERS).plus +
+                       closure("." + GRAPHEMES.plus, 1, 2))
+
+        time = (closure(NUMBERS, 1, 2) +
+                closure(":" + closure(NUMBERS, 1, 2), 1, 2))
+
+        fancy_numbers = (closure(NUMBERS, 1, 6) |
```

Comment:
Could you add a comment here to explain what types of numbers this is accepting, as this is a bit difficult to parse :)

Reply:
Added some comments explaining the ways these group numbers, with some examples.